app/code/Magento/MediaStorage/Model/File/Storage/File.php

```diff
@@ -291,16 +291,16 @@ public function saveFile($file, $overwrite = true)
             && isset($file['content'])
             && !empty($file['content'])
         ) {
-            try {
-                $filename = isset(
-                    $file['directory']
-                ) && !empty($file['directory']) ? $file['directory'] . '/' . $file['filename'] : $file['filename'];
+            $filename = isset(
+                $file['directory']
+            ) && !empty($file['directory']) ? $file['directory'] . '/' . $file['filename'] : $file['filename'];
```

Comment:
I think this one is a bit more readable. What do you think?
```suggestion
            $directory = !empty($file['directory'] ?? '') : $file['directory'] . '/' : '';
            $filename = $directory . $file['filename'];
```